tests/auth_tests/test_forms.py

```diff
@@ -261,6 +268,32 @@ class Meta(UserCreationForm.Meta):
             ['The password is too similar to the first name.'],
         )
 
+    @override_settings(AUTH_USER_MODEL='auth_tests.ExtensionUser')
+    def test_with_custom_user_model(self):
+        data = {
+            'username': 'test_username',
+            'password1': 'test_password',
+            'password2': 'test_password',
+        }
+        reload(django.contrib.auth.forms)
+        from django.contrib.auth.forms import UserCreationForm
```

Comment:
I'm a bit worried about this `reload()` with overridden settings to trigger the `UserCreationForm` to be recreated with a custom model. It obviously works for this test case, but I think it will leave the customised `UserCreationForm` loaded at the end of the test, won't it? Doesn't this mean that we will potentially have dependencies between this test and other ones that also use the `UserCreationForm`? If so, I think we should find a way to do a `reload()` at the end of the test (in `tearDown()`?) after the settings have been put back to the default.

Reply:
Yes, this code looks a bit suspicious. Perhaps a signal handler that listens for `setting_changed` might be appropriate.

Reply:
Maybe using context manager to override settings and reload the form is a good choice too? like this:
```
def test_customer_user_model_with_different_username_field(self):
	with override_settings(AUTH_USER_MODEL='auth_tests.CustomUser'):
	    reload(django.contrib.auth.forms)
	    from django.contrib.auth.forms import UserChangeForm
	    self.assertEqual(UserChangeForm.Meta.model, CustomUser)
	    # ...
	reload(django.contrib.auth.forms)
	from django.contrib.auth.forms import UserCreationForm
	self.assertEqual(UserCreationForm.Meta.model, User)
```
I have update it in the new commit.

Reply:
The signal approach would hopefully eliminate the need to call `reload()` manually.